riak_test/yz_solr_upgrade_downgrade.erl

```diff
@@ -0,0 +1,459 @@
+    rt:pmap(
+        fun({Bucket, Index}) ->
+            ok = write_and_verify_data(Cluster, Bucket, Index, 201, 100, 300)
+        end,
+        [{?BUCKET_A2, ?INDEX_A2},
+         {?BUCKET_B1, ?INDEX_B1},
+         {?BUCKET_B2, ?INDEX_B2},
+         {?BUCKET_B3, ?INDEX_B3},
+         {?BUCKET_C,  ?INDEX_C}]
+    ),
+    pass.
+
+write_data(Cluster, Bucket, Start, End) ->
+    ValueGenerator = fun(_Key) ->
+        {"Same as it ever was", "text/plain"}
+    end,
+    lists:all(
+        fun(Result) -> Result =:= ok end,
+        yz_rt:write_data(Cluster, Bucket, Start, End, ValueGenerator)
+    ).
+
+
+verify_data(Cluster, Index, NumKeys) ->
+    ok = yz_rt:verify_num_match(yokozuna, Cluster, Index, NumKeys),
+    ExpectedSolrMatch = (?N_VAL * NumKeys * length(Cluster)) div ?CLUSTER_SIZE,
+    ok = yz_rt:verify_num_match(solr, Cluster, Index, ExpectedSolrMatch),
+    ok.
+
+augment_config(ApplicationKey, Key, Value, Config) ->
+    ApplicationConfig = proplists:get_value(ApplicationKey, Config),
+    ApplicationDict = dict:from_list(ApplicationConfig),
+    NewApplicationConfig = dict:to_list(dict:store(Key, Value, ApplicationDict)),
+    lists:keyreplace(ApplicationKey, 1, Config, {ApplicationKey, NewApplicationConfig}).
+
+
+update_yz(Params) ->
+    NewDataDir = proplists:get_value(new_data_dir, Params),
+    TimestampStr = timestamp_str(),
+    %%
+    %% Move yz_temp out of the way, because it contains cached Solr JAR files
+    %% and other artifacts.
+    %%
+    ok = mv_yz_temp(NewDataDir, TimestampStr),
+    %%
+    %% Update the luceneMatchVersion in the Solr config
+    %%
+    [ok = modify_solr_config(NewDataDir, binary_to_list(Index),
+        TimestampStr, ?LUCENE_MATCH_4_10_4_VERSION) ||
+            Index <- [?INDEX_B1, ?INDEX_B2, ?INDEX_B3]
+    ],
+    %%
+    %% Move the data directories for indices B2 and B3 out of the way
+    %% (We will use the backup of B3 during downgrade)
+    %%
+    [mv_yz_index_data_dir(NewDataDir, binary_to_list(Index), TimestampStr)  ||
+        Index <- [?INDEX_B2, ?INDEX_B3]],
+    mv_yz_aae_dir(NewDataDir, TimestampStr),
+    ok.
+
+mv_yz_temp(DataDir, TimestampStr) ->
+    lager:info("Moving yz_temp directory out of the way..."),
+    YZTempPath = io_lib:format("~s/yz_temp", [DataDir]),
+    ok = file:rename(YZTempPath, io_lib:format("~s-~s", [YZTempPath, TimestampStr])).
+
+mv_yz_index_data_dir(DataDir, IndexName, TimestampStr) ->
+    lager:info("Moving yz data directory for index ~s out of the way...", [IndexName]),
+    YZTempPath = io_lib:format("~s/yz/~s/data", [DataDir, IndexName]),
+    ok = file:rename(YZTempPath, io_lib:format("~s-~s", [YZTempPath, TimestampStr])).
+
+mv_yz_aae_dir(DataDir, TimestampStr) ->
+    lager:info("Moving yz AAE directory out of the way..."),
+    YZTempPath = io_lib:format("~s/yz_anti_entropy", [DataDir]),
+    ok = file:rename(YZTempPath, io_lib:format("~s-~s", [YZTempPath, TimestampStr])).
+
+timestamp_str() ->
```

Comment:
Maybe put this in the `rt` module, in case someone else wants to use it for another test in the future?